### PR TITLE
mark channel and percolatequery is deleted and update memeberships

### LINKED ISSUE #4168: delete a channel
As an admin, I'd like to be able to delete a channel that I've created. 

In the past, we've done this ad hoc by deleting the `Channel` object associated with the channel. Unfortunately this has left thousands of "dangling" `PercolateQueryMemberships` that we attempt to synchronize with the discussions application every few minutes. 

- [ ] Add `is_deleted` columns to `Channel` and  `PercolateQuery` (`default=False`)
- [ ] Add a command that takes a channel name and sets the above fields to `True` and updates all the memberships to `is_member=False, need_update=True`
- [ ] Kicks off a task to remove moderators
- [ ] Update any areas where we query for `Channel` and `PercolateQuery` to filter out deleted ones

### Questions

- Should this remove memberships on the discussions side? (if yes, we may want to instead have a walk to mark the channel as deleted so the updates can still occur)
  - Yes, we need to clean this up too

### PR DESCRIPTION
**Pull request by tasawernawaz**
#### What are the relevant tickets?
Fixes #4168 

#### What's this PR do?

- [x]  Add is_deleted columns to `Channel` and `PercolateQuery` (default=False)
- [x]  Add a command that takes a channel name and sets the above fields to True and updates all the memberships to `is_member=False`, `need_update=True`
- [x]  Kicks off a task to remove moderators
- [x]  Update any areas where we query for Channel and PercolateQuery to filter out deleted ones
- [x] Add unit tests

### CODE REVIEW

**Comment by tasawernawaz** [comment]
@pdpinch what is and how we have defined the relation between moderators and channels ?

**Comment by tasawernawaz** [comment]
@rhysyngsun I guess @pdpinch is busy in OpenEdx conference, could you tell me that is there any way to get all moderators for a given channel in MM, as I have to delete them on deletion on channel.

I can see a relation in model `ChannelGroupRole` in Open-Discussion but not in MM.

**Comment by tasawernawaz** [comment]
@rhysyngsun thanks for answering the above questions. I have added code, could you please take a look, meanwhile I can add few tests.

File: discussions/tasks.py [hunk 1]
```diff
@@ -129,6 +131,32 @@ def add_user_as_moderator_to_channel(user_id, program_id):
...
 
+@app.task()
+def remove_moderators_from_channel(channel_name):
+    """
+    Remove moderators of a given channel_name.
+    Args:
+        channel_name(str): channel name
+    """
+
+    mod_ids = Role.objects.filter(
```
Comment (rhysyngsun) [inline discussions/tasks.py]: Can you refactor this and the usage in the  other method to a `get_moderators_for_channel` function?

**Comment by pdpinch** [comment]
@mbertrand Now that #4294 is merged can we resume the review of this? (not urgent)

**Comment by tasawernawaz** [comment]
@pdpinch I was about to comment on this one.

**Comment by pdpinch** [comment]
Matt and I spoke. He opened https://github.com/mitodl/open-discussions/issues/1971 to address the additional work in open, but this is fine to merge for now. It will address an recurring issue that we have with processing nonexistent channels.

**Pull request merged**